### PR TITLE
Some locales need to be converted to be compliant with CLDR

### LINKED ISSUE #16207: Serbian language error (sr@latin)
<h3>Steps to reproduce</h3>

1. choose Srpski as preferred language

<h3>Expected behaviour</h3>

Application should work

<h3>Actual behaviour</h3>

Application is unusable with these errors in the server log file:
<pre><code>
Exception: {"Message":"'sr@latin' is not a valid locale identifier","Code":10001,"Trace":"#0 \/var\/www\/html\/owncloud\/3rdparty\/punic\/punic\/code\/Data.php(481): Punic\\Data::getLocaleAlternatives('sr@latin')\n#1 \/var\/www\/html\/owncloud\/3rdparty\/punic\/punic\/code\/Data.php(124): Punic\\Data::getLocaleFolder('sr@latin')\n#2 \/var\/www\/html\/owncloud\/3rdparty\/punic\/punic\/code\/Calendar.php(647): Punic\\Data::get('calendar', 'sr@latin')\n#3 \/var\/www\/html\/owncloud\/lib\/private\/l10n.php(519): Punic\\Calendar::getDateFormat('short', 'sr@latin')\n#4 \/var\/www\/html\/owncloud\/core\/js\/config.php(42): OC_L10N->getDateFormat()\n#5 \/var\/www\/html\/owncloud\/lib\/private\/route\/route.php(135) : runtime-created function(1): require_once('\/var\/www\/html\/o...')\n#6 [internal function]: __lambda_func(Array)\n#7 \/var\/www\/html\/owncloud\/lib\/private\/route\/router.php(250): call_user_func('\\x00lambda_1961', Array)\n#8 \/var\/www\/html\/owncloud\/lib\/base.php(782): OC\\Route\\Router->match('\/core\/js\/oc.js')\n#9 \/var\/www\/html\/owncloud\/index.php(36): OC::handleRequest()\n#10 {main}","File":"\/var\/www\/html\/owncloud\/3rdparty\/punic\/punic\/code\/Data.php","Line":507}
</code></pre>

<h3>Server Host info:</h3>

Ubuntu 12.04
PHP 5.5.9

<h3>Server configuration:</h3>

ownCloud Enterprise Edition 8.0.3

### PR DESCRIPTION
**Pull request by DeepDiver1975**
fixes  #16207 
@UrosStankovic please test - THX

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
fixes the issue as commented by the reporter https://github.com/owncloud/core/issues/16207#issuecomment-100494323 - THX @UrosStankovic for testing

**Comment by MorrisJobke** [comment]
Works :+1: (error is gone)

File: lib/private/l10n.php [hunk 1]
```diff
@@ -546,6 +548,15 @@ public function getDateFormat() {
 	 */
 	public function getFirstWeekDay() {
 		$locale = $this->getLanguageCode();
+		$locale = $this->transformToCLDRLocale($locale);
 		return Punic\Calendar::getFirstWeekday($locale);
 	}
+
+	private function transformToCLDRLocale($locale) {
```
Comment (LukasReschke) [inline lib/private/l10n.php]: One shall provide PHPDocs even for private functions :see_no_evil:

**Comment by LukasReschke** [comment]
:+1:

File: lib/private/l10n.php (hunk 1 above)
Comment (DeepDiver1975) [inline lib/private/l10n.php]: > One shall provide PHPDocs even for private functions :see_no_evil:

<<-- that one will do with the next PR - there are more languages we have to properly map ...

**Pull request merged**